### ISSUE TITLE
Support macro expansion / xattr

### ISSUE THREAD

**Issue by daschl** [couchbase/CouchbaseMock#46]
Hi,

one example failed against the mock: I'm trying to cover the case "get with expiration" so it would have to fall back to subdoc. I do a XATTR GET to `$document.exptime` combined with a GET_DOC which would give me all I need... this works fine against a real server but fails against the mock.

I assume it is because the mock is missing xattr/macro expansions?

**Comment by avsej** [comment]
@daschl, yes, you right. Mock does not support macros yet, but I will fix that soon.
